### ISSUE TITLE
Add uniform spacing properties to panels

### ISSUE THREAD

**Issue by harinikmsft** [Microsoft/xaml-standard#9]
- Spacing on StackPanel (uniform distance between children in the direction of orientation)

- RowSpacing and ColumnSpacing on Grid (uniform spacing between row and column items)

```xml
<StackPanel Orientation="Vertical" Spacing="10">
   <!-- 10 pixel distance vertically between the text and the button below -->
   <TextBlock Text="Hello world!" />
   <Button Content="Click me" />
</StackPanel>

<Grid RowSpacing="10" ColumnSpacing="5">
   <Grid.RowDefinitions>
      <RowDefinition Height="100" />
      <RowDefinition Height="200" />
   </Grid.RowDefinitions>
   <Grid.ColumnDefinitions>
      <ColumnDefinition Width="100" />
      <ColumnDefinition Width="200" />
   </Grid.ColumnDefinitions>
   <!-- 10 pixels distance between the 2 textblocks below -->
   <TextBlock Text="Name: " />
   <TextBlock Text="Address: " Gris.Row="1" /> 
   <!-- 5 pixels distance between each TextBlock above and its corresponding row's TextBox below -->
   <!-- And 10 pixels distance between the 2 texgboxes below -->
   <TextBox Grid.Column="1" />
   <TextBox Grid.Column="1" Grid.Row="1" />
</Grid>

**Comment by h82258652** [comment]
But use ```Margin``` in child element also can do this, I dont think it is necessary.

**Comment by billreiss** [comment]
I like this idea, it's very handy in Xamarin Forms and cleaner than putting margins everywhere. You could always still use margins if you want to but this is a great option.

**Comment by nickrandolph** [comment]
I agree with @h82258652 that whilst I like the concept, this shouldn't make v1 of the standard. It's something that's easily achievable with Margins

**Comment by birbilis** [comment]
The difference from Margins is that you put it in the parent. Could probably be done with Attachable Properties, aka set once in the parent and have the parent override (not set) the Margin in the children. Don't remember though if you can attach properties like that to override existing ones. 

Or just add to the child margins the value of the spacing property (1 or 2 or 4 values) when laying out. It would default to 0 of couse

**Comment by nickrandolph** [comment]
I think there needs to be a balance in v1 of the standard between providing all the essentials that XAML developers need and over engineering the first version. Better to keep to the essentials that make sense for the widest audience.

**Comment by PedroLamas** [comment]
Might I just point out that [Grid.ColumnSpacing](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.grid#Windows_UI_Xaml_Controls_Grid_ColumnSpacing), [Grid.RowSpacing](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.grid#Windows_UI_Xaml_Controls_Grid_RowSpacing), [StackPanel.Spacing](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.stackpanel#Windows_UI_Xaml_Controls_StackPanel_Spacing) are now on UWP preview, so it might actually make sense to bring this to the standard, as it is already supported by Xamarin XAML anyway!
